### PR TITLE
add signup.ejs and a signup button in sidebar.ejs

### PR DESCRIPTION
**Pull request by theCodeCrusaderX**
opensource0

as it my first contribution i want some feedback from u ,

i did not able to add user using postman and even using browser as it is restricted by u .

i have add the image that what exactly i am getting while registering the user .

is this ok or i need or i implemented wrong 

waiting for your feedback

### CODE REVIEW

**Comment by pooranjoyb** [comment]
lemme check the permissions for mongodb.

Edit: 
Please check the APIs now, I've updated the permissions for user `hacktoberfest` @theCodeCrusaderX

**Comment by theCodeCrusaderX** [comment]
yupp!! signup is working well from my end 



once the user click signup button the user info is being stored in db check that one 

one more thing do i need to protect all the route as till now it only signup the user

File: views/pages/Signup.ejs
Comment (pooranjoyb) [inline views/pages/Signup.ejs:1]: okay so there are few changes that you need to make.

- Use packages like [Toastify](https://apvarun.github.io/toastify-js/#) for response messages.
- Donot redirect and show response like this.

image

- Create the SignUp redirection in the Avatar itself and NOT a separate button.

image

- Also make sure you maintain the Theme and Color, Please have a look onto this, please.

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

File: views/pages/Signup.ejs
Comment (theCodeCrusaderX) [reply views/pages/Signup.ejs:1]: Ok thanks for feedback 😃

**Comment by theCodeCrusaderX** [comment]
did the changes that you wnated have a look from your side too...

completed signup form .

**Comment by theCodeCrusaderX** [comment]
Please check it so then I will work on login form too...

File: views/includes/sidebar.ejs [hunk 1]
```diff
@@ -2,4 +2,7 @@
     <a href="/profile">
         <img style="border-radius: 100%; width: 3rem;" src="/images/avatar.jpg" alt="">
     </a>
+    <a href="/signup">
+        <button class="p-3 rounded-lg border">Signup</button>
```
Comment (pooranjoyb) [inline views/includes/sidebar.ejs:6]: Remove this button as we are navigating using the Avatar icon.

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

File: views/pages/profile.ejs [hunk 2]
```diff
@@ -1,45 +1,151 @@
 <!DOCTYPE html>
 <html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>About</title>
```
Comment (pooranjoyb) [inline views/pages/profile.ejs:6]: Change this to Profile

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

**Comment by pooranjoyb** [comment]
I guess that's all, Probably ready to merge. @theCodeCrusaderX

**Pull request merged**